src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerManager.AnalyzerExecutionContext.cs

```diff
@@ -20,7 +20,7 @@ private sealed class AnalyzerExecutionContext
 
             // This map stores the tasks to compute HostCompilationStartAnalysisScope for per-compilation analyzer actions, i.e. AnalyzerActions registered by analyzer's CompilationStartActions.
             // Compilation start actions will get executed once per-each AnalyzerAndOptions as user might want to return different set of custom actions for each compilation/analyzer options.
```

Comment:
I am going to delete a bunch of code around this in dev15.6 branch soon, we are going to get rid of the singleton.